runtime/v2/manager.go

```diff
@@ -854,7 +861,7 @@ func (s *stfRouterWrapper) RegisterHandler(handler appmodulev2.Handler) {
 	s.error = errors.Join(s.error, err)
 
 	// also make the decoder
-	if s.error == nil {
+	if s.error != nil {
```

Comment:
shall we just check if  `s.handlers` is nil? Checking the error doesn't seem related tbh